public/app/features/alerting/unified/mocks/server/db.ts

```diff
@@ -0,0 +1,79 @@
+import { Factory } from 'fishery';
+
+import { DataSourceInstanceSettings, PluginType } from '@grafana/data';
+import { config, setDataSourceSrv } from '@grafana/runtime';
+import {
+  PromAlertingRuleDTO,
+  PromAlertingRuleState,
+  PromRuleGroupDTO,
+  PromRuleType,
+} from 'app/types/unified-alerting-dto';
+
+import { MockDataSourceSrv } from '../../mocks';
+import { DataSourceType } from '../../utils/datasource';
+
+const ruleFactory = Factory.define<PromAlertingRuleDTO>(({ sequence }) => ({
+  name: `test-rule-${sequence}`,
+  query: 'test-query',
+  state: PromAlertingRuleState.Inactive,
+  type: PromRuleType.Alerting,
+  health: 'ok',
+  labels: { team: 'infra' },
+}));
+
+const groupFactory = Factory.define<PromRuleGroupDTO>(({ sequence }) => {
+  const group = {
+    name: `test-group-${sequence}`,
+    file: `test-namespace`,
+    interval: 10,
+    rules: ruleFactory.buildList(10),
+  };
+
+  ruleFactory.rewindSequence();
+
+  return group;
+});
+
+const dataSourceFactory = Factory.define<DataSourceInstanceSettings>(({ sequence, params, afterBuild }) => {
```

Comment:
This has a fair overlap with our `mockDataSource` method (its a bit more limited than that method, actually). We should combine the two (I have some thoughts/a WIP branch to tidy up some datasource logic as well)

Reply:
This is mostly to see how mocking could look like with `fishery`. More like a starting point to discuss if we want to go this way or not